services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/deviceconnection/MapBasedDeviceConnectionService.java

```diff
@@ -159,13 +171,13 @@ public long expireAfterCreate(final String key, final ExpiringValue<JsonObject>
                     @Override
                     public long expireAfterUpdate(final String key, final ExpiringValue<JsonObject> value,
                             final long currentTime, final long currentDuration) {
-                        return Long.MAX_VALUE;
+                        return value.getLifespanNanos();
                     }
 
                     @Override
                     public long expireAfterRead(final String key, final ExpiringValue<JsonObject> value,
                             final long currentTime, final long currentDuration) {
-                        return Long.MAX_VALUE;
+                        return currentDuration;
```

Comment:
My understanding is that we do not want to change an existing value's lifespan on reading, right? Does *currentDuration* contain the object's *remaining* lifetime based on the lifetime determined at creation/update time?

Reply:
Yes. As the javadoc states: `The {@code currentDuration} may be returned to not modify the expiration time.`. So using `Long.MAX_VALUE` was wrong here.